apps/mobile/src/components/ModifyOffer/atoms/offerFormStateAtoms.ts

```diff
@@ -112,14 +112,17 @@ export const offerFormMolecule = molecule(() => {
     null,
     (get, set, params) => {
       const {currency} = params
-      set(currencyAtom, currency)
-      set(amountBottomLimitAtom, get(amountBottomLimitUsdEurCzkAtom))
-      set(
-        amountTopLimitAtom,
-        currency === 'CZK'
-          ? get(amountTopLimitCzkAtom)
-          : get(amountTopLimitUsdEurAtom)
-      )
+      const currencyFromAtom = get(currencyAtom)
+      if (currencyFromAtom !== currency) {
```

Comment:
It's nice practice to actually invert such conditions to make the code more readable and prevent unnecessary nesting and the pyramid of doom [see this](https://www.youtube.com/watch?v=CFRhGnuXG-4)

in this case that would be: 
`if(previousCurrency === newCurrency) return`

Reply:
@kaladivo good point! thx